packages/x-date-pickers/src/dateViewRenderers/dateViewRenderers.tsx

```diff
@@ -1,12 +1,15 @@
 import * as React from 'react';
 import { DateCalendar, DateCalendarProps } from '../DateCalendar';
 import { DateOrTimeView, DateView } from '../models';
+import { DateOrTimeViewWithMeridiem } from '../internals/models';
 
 const isDatePickerView = (view: unknown): view is DateView =>
   view === 'year' || view === 'month' || view === 'day';
 
-export interface DateViewRendererProps<TDate, TView extends DateOrTimeView>
-  extends Omit<
+export interface DateViewRendererProps<
+  TDate,
+  TView extends DateOrTimeViewWithMeridiem | DateOrTimeView,
```

Comment:
This change is not necessary yet, because we are not adding the `digital clock` to DateTimePicker YET. 🤔 
But IMHO it makes sense to refactor all places where it will be needed and it doesn't hurt now. 🤷